### PR TITLE
Fix Prometheus Pushgateway can not enable

### PR DESCRIPTION
**Pull request by FoghostCn**
Fix Prometheus Pushgateway can not enable

### CODE REVIEW

File: metrics/options.go [hunk 1]
```diff
@@ -76,6 +76,13 @@ func WithPrometheusExporterEnabled() Option {
 	}
 }
 
+func WithPrometheusPushgatewayEnabled() Option {
+	return func(opts *Options) {
+		enabled := true
+		opts.Metrics.Prometheus.Pushgateway.Enabled = &enabled
```
Comment (baerwang) [inline metrics/options.go:82]: why use pointer?

Comment (FoghostCn) [reply metrics/options.go:82]: cause https://github.com/apache/dubbo-go/issues/2335

Comment (baerwang) [reply metrics/options.go:82]: There are multiple PRs in this issue that have resolved this issue

Comment (FoghostCn) [reply metrics/options.go:82]: Yes,and this pr just add a method to modify the configuration

**Comment by justxuewei** [comment]
Could you backport this patch to `release-3.1`?

**Review by justxuewei — CHANGES_REQUESTED** [review]
Block temporarily.

**Comment by FoghostCn** [comment]
> Could you backport this patch to `release-3.1`?

`release-3.1` has no options support,so this pr no need to backport to `release-3.1`

**Review by justxuewei — APPROVED** [review]
Lgtm！

**Pull request merged**